staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,340 @@
+	// if not specified, it will be defaulted to 8.
+	// +optional
+	HandSize int32 `json:"handSize" protobuf:"varint,3,opt,name=handSize"`
+	// `queueLengthLimit` must be a length limit applied to each queue belongs to the priority.  The value must be positive
+	// for a non-exempt priority level.
+	// if not specified, it will be defaulted to 50.
+	// +optional
+	QueueLengthLimit int32 `json:"queueLengthLimit" protobuf:"varint,4,opt,name=queueLengthLimit"`
+	// `exempt` defines whether the priority level is exempted or not.  There should be at most one exempt priority level.
+	// Being exempt means that requests of that priority are not subject to concurrency limits (and thus are never queued)
+	// and do not detract from the concurrency available for non-exempt requests. The field is default to false and only those system
+	// preset priority level can be exempt.
+	// if not specified, defaulted to non-exempt i.e. false.
+	// +optional
+	Exempt bool `json:"exempt,omitempty" protobuf:"varint,5,opt,name=exempt"`
+}
+
+// PriorityLevelConfigurationConditionType is a valid value for PriorityLevelConfigurationStatusCondition.Type
+type PriorityLevelConfigurationConditionType string
+
+// PriorityLevelConfigurationStatus represents the current state of a "request-priority".
+type PriorityLevelConfigurationStatus struct {
+	// `conditions` is the current state of "request-priority".
+	Conditions []PriorityLevelConfigurationCondition `json:"conditions,omitempty" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+// PriorityLevelConfigurationCondition defines the condition of priority level.
+type PriorityLevelConfigurationCondition struct {
+	// `type` is the type of the condition.
+	// Required.
+	Type PriorityLevelConfigurationConditionType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type"`
+	// `status` is the status of the condition.
+	// Can be True, False, Unknown.
+	// Required.
+	Status ConditionStatus `json:"status,omitempty" protobuf:"bytes,2,opt,name=status"`
+	// `lastTransitionTime` is the last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,3,opt,name=lastTransitionTime"`
+	// `reason` is a unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// `message` is a human-readable message indicating details about last transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+
+// Subject matches a set of users.
+// Syntactically, Subject is a general API object reference.
+// Authorization produces a username and a set of groups, and we imagine special kinds of non-namespaced objects,
+// User and Group to represent such a username or group.
+// The only kind of true object reference that currently will match any users is ServiceAccount.
+type Subject struct {
```

Comment:
Let's move this up to the flow schema section, perhaps just after it is referenced.

Reply:
moved